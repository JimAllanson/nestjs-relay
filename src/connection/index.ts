export * from './backward-connection-args.type';
export * from './connection-args.type';
export * from './connection.type';
export * from './forward-connection-args.type';
export * from './page-info.type';
export * from './resolve-connection-field.decorator';
