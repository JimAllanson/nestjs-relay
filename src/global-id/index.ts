export * from './global-id-field.resolver'
export * from './global-id.scalar'
export * from './resolved-global-id.type'
