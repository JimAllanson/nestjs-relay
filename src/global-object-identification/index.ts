export {
  GlobalId,
  GlobalIdField,
  GlobalIdFieldOptions,
  GlobalIdFieldResolver,
  GlobalIdScalar,
  ResolvedGlobalId,
} from './global-id';

export { NodeFieldResolver, NodeInterface, NodeResolver } from './node';
