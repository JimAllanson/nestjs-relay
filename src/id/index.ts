export * from './global-id.type'
export * from './id.scalar'
