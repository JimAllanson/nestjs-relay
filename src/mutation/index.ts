export { RelayArg, RelayArgOptions } from './relay-arg.decorator';
export { RelayArgs, RelayArgsOptions } from './relay-args.decorator';
