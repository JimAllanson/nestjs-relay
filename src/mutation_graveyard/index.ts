export * from './relay-mutation.decorator';
