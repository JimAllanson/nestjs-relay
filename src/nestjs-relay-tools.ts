export * from './id'
export * from './node'
