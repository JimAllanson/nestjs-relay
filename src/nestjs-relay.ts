export * from './global-id';
export * from './mutation_graveyard';
export * from './node';
