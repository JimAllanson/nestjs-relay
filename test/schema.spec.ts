import { NestFactory } from '@nestjs/core'
import { GraphQLSchemaBuilderModule, GraphQLSchemaFactory } from '@nestjs/graphql'
import { GraphQLSchema, printSchema } from 'graphql'
import { IDScalar } from '../src/nestjs-relay-tools'
import { FactionResolver, NodeFieldResolver } from './starwars-app'

const resolvers = [FactionResolver, NodeFieldResolver]
const scalars = [IDScalar]

describe('Schema', () => {
  let schema: GraphQLSchema

  beforeAll(async () => {
    const app = await NestFactory.create(GraphQLSchemaBuilderModule)
    await app.init()

    const gqlSchemaFactory = app.get(GraphQLSchemaFactory)
    schema = await gqlSchemaFactory.create(resolvers, scalars)
  })

  it('should contain the `Node` interface type', () => {
    const nodeType = schema.getType('Node')
    expect(nodeType).toBeDefined()
  })

  it('should contain the `node` field on the `Query` type', () => {
    const queryType = schema.getQueryType()
    expect(queryType).toBeDefined()

    const nodeField = queryType?.getFields()['node']
    expect(nodeField).toBeDefined()
  })

  it('should match snapshot', () => {
    expect(printSchema(schema)).toMatchSnapshot()
  })
})
