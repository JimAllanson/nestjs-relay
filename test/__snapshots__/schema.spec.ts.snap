// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema should match snapshot 1`] = `
"\\"\\"\\"An object with an ID\\"\\"\\"
interface Node {
  \\"\\"\\"The ID of the object\\"\\"\\"
  id: ID!
}

type Faction implements Node {
  id: ID!
  name: String
}

type Ship implements Node {
  id: ID!
  name: String
}

type Query {
  rebels: Faction
  empire: Faction

  \\"\\"\\"Fetches an object given its ID\\"\\"\\"
  node(
    \\"\\"\\"The ID of an object\\"\\"\\"
    id: ID!
  ): Node
}
"
`;
